benchmarks/infra/stage-2/modules/gke-setup/modules/workload-identity/kubernetes.tf

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  wi_sa_email = (
+    var.google_service_account_create
+    ? module.workload-service-account.0.email
+    : data.google_service_account.gsa.0.email
+  )
+  wi_sa_name = (
+    var.google_service_account_create
+    ? module.workload-service-account.0.name
+    : data.google_service_account.gsa.0.name
+  )
+}
+
+resource "kubernetes_namespace" "namespace" {
+  count = var.namespace_create ? 1 : 0
+  metadata {
+    annotations = {
+      name = var.namespace
+    }
+    name = var.namespace
+  }
+}
+
+resource "kubernetes_service_account" "ksa" {
```

Comment:
what is this ksa used for? please just use the TF module for WI instead of doing all the steps manually: https://registry.terraform.io/modules/terraform-google-modules/kubernetes-engine/google/latest/submodules/workload-identity

Reply:
The KSA's are used for accessing related resources for benchmarking: model weights, benchmarking datasets, and output datasets. Each one has their own KSA for isolation.

+1, I think it makes sense for us to clean this up as part of the refactor onto the terraform-google-modules.

Reply:
Yes, we will follow up when we refactor.